src/components/Form/ExitFormDialogProvider.tsx

```diff
@@ -36,21 +56,63 @@ const ExitFormDialogProvider = ({ children }) => {
 
   const [showDialog, setShowDialog] = useState(defaultValues.showDialog);
 
-  const submitRef = useRef<() => SubmitPromise<boolean>>(null);
+  const isSubmitting = useRef(defaultValues.isSubmitting);
+  const formsData = useRef<FormsData>({});
   const blockNav = useRef(defaultValues.blockNav);
   const navAction = useRef(defaultValues.navAction);
-  const isDirty = useRef(defaultValues.isDirty);
   const enableExitDialog = useRef(defaultValues.enableExitDialog);
+  const currentPath = useRef(window.location.pathname);
+
+  // Set either on generic form load or on every custom form data change
+  // but doesn't cause re-renders
+  function setSubmitRef<T extends () => SubmitPromise<any[]>>(
```

Comment:
Why is it a function expression while the rest uses variable functions? Move it under setFormData definition, so it's easier to read top-to-bottom